webapp/app/forms/flows/step_chooser.py

```diff
@@ -57,9 +60,25 @@ def get_correct_step(self, step_name: str, update_data: bool = False) -> Steuerl
             overview_step=self.overview_step,
             prev_step=self.determine_prev_step(step_name, stored_data),
             next_step=self.determine_next_step(step_name, stored_data),
-            session_data_identifier=self.session_data_identifier
+            session_data_identifier=self.session_data_identifier,
+            update_data=update_data,
+            data_is_valid=data_is_valid is not None,
+            form_data=form_data
         )
 
+    def validate_and_update_data(self, step_name: str, update_data: bool, stored_data, form_data: ImmutableMultiDict = None):
+        """
+        :param step_name: The name of the step. Should be a valid step name
+        """
+        data_is_valid = None
+        if update_data:
+            if validated_data := self.steps[step_name].validate_data(form_data, stored_data):
+                stored_data = self.steps[step_name].update_data(stored_data, validated_data)
+
+            data_is_valid = validated_data is not None
+
+        return data_is_valid, stored_data
```

Comment:
I don't think this checking for `is not None` works for me. neither here nor in l. 65.
My suggestion - and I can't really believe I'm saying that - is to throw an exception if the form doesn't validate and catch that by setting data_is_valid to False.

Reply:
You mean that we would throw an error in line 78, catch that in line 54 and set data_is_valide to False in the except block?

Reply:
Mh no, I think I meant throwing the error within `vlaidate_data` instead of returning None.

Reply:
I have changed that quite a lot. Maybe you can have a look if you still need to change how it is implemented?